lib/train/extras/command_wrapper.rb

```diff
@@ -162,7 +162,13 @@ class CommandWrapper
     include_options WindowsCommand
 
     def self.load(transport, options)
-      if transport.os.unix?
+      if transport.class == Train::Transports::SSH::Connection && !transport.connected?
+        # Need to call run_command here to get the message and show it to user while raising exception
+        msg = transport.run_command("whoami").stdout
+        # this will throw the error as
+        # Train::UserError: Ssh failed: Please login as the user "username" rather than the user "root".
+        raise Train::UserError.new("Ssh failed: #{msg}")
```

Comment:
A better (more rubyish) way to say this is: `raise Train::UserError, "Ssh failed: #{msg}"`